halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/config/v1/secrets/DecryptingObjectMapper.java

```diff
@@ -49,6 +49,7 @@
  * <p>decryptedOutputDirectory is the path to the decrypted secret files on the service's host.
  */
 public class DecryptingObjectMapper extends ObjectMapper {
+  private static final String K8S_SECRET_ENGINE = "k8s";
```

Comment:
As this is intended to decrypt a specific secret engine.
I'm wondering if we should do the same for different ones (aka s3, vault).
Or is there any specific behavior that just applied for k8s?

Reply:
Spinnaker services have the ability to decrypt secrets from all other engines (S3, GCS and Vault) on their own, by means of including kork or go-yaml-tools libraries.

Kubernetes engine is different because operator is the only one with the ability to decrypt those secrets. When Spinnaker is deployed, k8s secrets get replaced by environment variables or mounted files created by the operator. Therefore, Halyard should never try to decrypt a k8s secret and operator should always decrypt them at deploy time.

Reply:
Gotcha, That makes a lot of sense now.
Thanks!